### DISCUSSION TITLE
Diarization

### DISCUSSION THREAD

**Discussion by laarisyko** [Stage-Whisper/Stage-Whisper#101] [Ideas]
I think Diarization is one of the more important features to realy make it user friendly. It would be great if it could even handle more than two speakers. I'm thinking about team meetings. Majdoddin has done a great job implementing it for two speakers https://github.com/Majdoddin/nlp

**Comment by petersterne** [comment]
Wow, that's a very cool approach! Do you know if we can run Pyannote locally (since we're running Whisper locally) or would we have to run it in the cloud?

**Reply by laarisyko** [reply]
I think so, however I'm not 100 % sure. I will ask Majdoddin.

**Reply by Majdoddin** [reply]
Sure, we can run both Pyannote and Whisper locally (needs GPU). Actually I had almost finished writing a python script of my work (v1), I can update it to (v2). v2 can already handle any number of speakers.

**Reply by Majdoddin** [reply]
We can run it on CPU efficiently, if we sparsify Whisper and Pyannote models with DeepSparse from [Neuralmagic](https://neuralmagic.com/). They have done it for BERT, so here it should work too.

**Reply by Majdoddin** [reply]
Meanwhile we can use die [Openvino model](https://github.com/openai/whisper/discussions/208) of @zhuzilin, at least for tiny-medium models. Because it is for arm architecture, I suspect it could be adapted to run on Android smartphones.

**Reply by laarisyko** [reply]
oh, this sounds great, if we could run BERT too. I have seen some scripts which used BERT to genreate summaries of a whisper transcript, which would be the ultimate added value for meeting transcriptions :)

**Reply by oelna** [reply]
I don't know whether this belongs here, but I just saw [openai-whisper-speaker-identification](https://github.com/zachlatta/openai-whisper-speaker-identification) by @zachlatta and I think identifying speakers in the transcripts would be an important feature.

**Reply by petersterne** [reply]
Thanks @oelna! I agree that speaker diarization is a crucial feature. We were looking into using PyAnnote, but I'm interested to try out @zachlatta's implementation of speaker diarization too.
